### ISSUE TITLE
Package Request: IPFS

### ISSUE THREAD

**Issue by phikal** [voidlinux/void-packages#3037]
I wanted to make a [IPFS](http://ipfs.io) package myself, so this is more of a question than a request. There would be three ways to create the package:

1. Use gobuilder.me, to download a prebuild version - not sure how well this goes in hand with XBPS though...
2. With go get, but that would get a dev package (I wanted to do this, but then I remembered #2614).
3. Download a [release](https://github.com/ipfs/go-ipfs/releases), and build it. I haven't looked too much into how this is done, but seems like the best way to do it.

What would be the most idiomatic way to go forth?

---

On a (un-)related side note: Could Void Linux host their installation images on ipfs? I'd imagine it would be easier than doing the same with BitTorrent (just `ipfs add [file]`), and it would make hosting easier. I believe Void would be (one again) one of the first distros to be the first to adapt newer/better technologies, so this would be fitting.

**Comment by dominikh** [comment]
1. gobuilder.me is not an option. We build our packages from sources whenever possible
2. go get is not an option. We package versioned software, not HEAD off of a VCS
3. Yes, download a release and build it. We have a build style for Go programs.

**Comment by 7185** [comment]
I'm also interested in this package.
This is what I got so far:
```
# Template file for 'go-ipfs'
pkgname=go-ipfs
version=0.3.9
revision=1
build_style=go
short_desc="Global versioned P2P merkle DAG file system"
license="MIT"
go_import_path="github.com/ipfs/go-ipfs"
homepage="https://$go_import_path"
distfiles="$homepage/archive/v$version.tar.gz"
checksum=74c8cac25f7ef0c5702fe7b59421b2072169ae357d393990c16cc7caccccbc73
```
But it didn't find any binaries, so I tried once more with:
```
do_install() {
	find "${GOPATH}/src/${go_import_path}/bin" -type f -executable | while read line
	do
		vbin "${line}"
	done
}
```
which was a bit better, but still no `ipfs` binary installed. I have no idea of what I am doing, but I feel that I'm not too far.

**Comment by CMB** [comment]
@7185 You are very close.  Add the line
go_package="${go_import_path}/cmd/ipfs"
just below go_import_path.
You'll also want to install the README.md and LICENSE in do_post_install.
I have a pull request ready for this package, or you can send it if you want.

**Comment by 7185** [comment]
@CMB Thank you, that's much better indeed. Feel free to submit yours since I'm not really good at this.
Also I think that it is possible to add some bash-completion file from the `misc/completion` directory.

**Comment by Duncaen** [comment]
Merged in 95a68fb
